### PR TITLE
Graphql doc

### PR DESCRIPTION
**Pull request by SamuelMartini**
Install [graphql-docs](https://github.com/gjtorikian/graphql-docs) to generate static html and markdown files from [schema.graphql](https://github.com/solidusio-contrib/solidus_graphql_api/blob/master/schema.graphql).

~This pr adds Type descriptions following a link to the official [solidus guide](https://guides.solidus.io/developers/). Descriptions are directly added to the file containing the Type ruby code as this is the best way to include the documentation inside the schema.graphql which can also serve the documention to any grapqhl clients.~

To generate the static files run `bundle exec rake schema:generate_docs`. However, being the doc fetched from the schema this rake task is also invoked after `bundle exec rake schema:idl` to ensure the doc is always updated.

**After merge**
Change github pages setting to fetch from master. Currently it is fetching from `gh-pages` branch.

**Documentation** can be found at https://solidusio-contrib.github.io/solidus_graphql_api/docs/

### CODE REVIEW

File: lib/graphql-docs/landing_pages/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- Check out GraphQL's [official documentation](https://graphql.org/)
+
+
+## Need help?
+
+Join the [SolidusSlack#support-channel](http://slack.solidus.io/) and start a conversation.
+
+## Queries
+
+1. `completedOrders` : List of customr completed orders.
```
Comment (ChristianRimondi) [inline lib/graphql-docs/landing_pages/index.md]: There is a typo in the `completedOrders` description (`customr`)

File: Rakefile [hunk 2]
```diff
@@ -33,6 +35,15 @@ namespace :schema do
 
     Rake::Task['graphql:schema:json'].invoke
   end
+
+  desc 'Generates documentation from schema.graphql'
```
Comment (ChristianRimondi) [inline Rakefile]: WDYT about removing the rake task from this commit?

File: schema.graphql [hunk 3]
```diff
@@ -852,7 +852,10 @@ type PageInfo {
 }
 
 """
-Payment.
+        The Spree::Payment model in Solidus tracks payments.
```
Comment (ChristianRimondi) [inline schema.graphql:855]: I'm not sure why there are changes to the `schema.graphql` in this commit 🤔

File: lib/solidus_graphql_api/types/credit_card.rb [hunk 4]
```diff
@@ -5,7 +5,12 @@ module Types
     class CreditCard < Base::RelayNode
       implements Types::Interfaces::PaymentSource
 
-      description 'Credit Card.'
+      description <<-INFO
+       If your payment processing integration uses the `Spree::CreditCard` class for its payment source,
+       take note that this model does not store all of the payment details.
+
+        More at [Solidus Guide](https://guides.solidus.io/developers/payments/payment-sources.html#credit-cards)
```
Comment (ChristianRimondi) [inline lib/solidus_graphql_api/types/credit_card.rb:12]: extra space detected! 🤖

File: lib/solidus_graphql_api/types/product_property.rb [hunk 5]
```diff
@@ -3,7 +3,11 @@
 module SolidusGraphqlApi
   module Types
     class ProductProperty < Base::RelayNode
-      description 'Product Property.'
+      description <<-INFO
+        Product properties belong to the `Spree::ProductProperty` model. They track individual attributes for a product.
```
Comment (rainerdema) [inline lib/solidus_graphql_api/types/product_property.rb:7]: .

File: lib/solidus_graphql_api/types/line_item.rb [hunk 6]
```diff
@@ -3,7 +3,12 @@
 module SolidusGraphqlApi
   module Types
     class LineItem < Base::RelayNode
-      description 'Line item.'
+      description <<-INFO
+        The `Spree::LineItem` model provides the cost of each item added to an order.
+        Line items provide a link between the order and Spree::Products and `Spree::Variants`.
```
Comment (ChristianRimondi) [inline lib/solidus_graphql_api/types/line_item.rb:8]: backticks on `Spree::Products` missing 🤖

File: lib/solidus_graphql_api/types/payment.rb [hunk 7]
```diff
@@ -3,7 +3,12 @@
 module SolidusGraphqlApi
   module Types
     class Payment < Base::RelayNode
-      description 'Payment.'
+      description <<-INFO
+        The Spree::Payment model in Solidus tracks payments.
```
Comment (ChristianRimondi) [inline lib/solidus_graphql_api/types/payment.rb:7]: backticks on `Spree::Payment` missing 🤖

File: lib/graphql_docs/landing_pages/index.md [hunk 8]
```diff
@@ -19,7 +19,7 @@ Join the [SolidusSlack#support-channel](http://slack.solidus.io/) and start a co
 
 ## Queries
 
-1. `completedOrders` : List of customr completed orders.
+1. `completedOrders` : List of customer completed orders.
```
Comment (rainerdema) [inline lib/graphql_docs/landing_pages/index.md:22]: move

Comment (rainerdema) [inline lib/graphql_docs/landing_pages/index.md]: This change should be move in the previous commit

File: solidus_graphql_api.gemspec [hunk 9]
```diff
@@ -35,6 +35,7 @@ Gem::Specification.new do |s|
   s.add_dependency 'solidus_core', ['>= 2.5', '< 3']
   s.add_dependency 'solidus_support', '~> 0.5'
 
+  s.add_development_dependency 'graphql-docs'
```
Comment (ChristianRimondi) [inline solidus_graphql_api.gemspec:38]: I think that it's better to "lock" the `graphql-docs` version (`~> 2.0.1`)

File: Rakefile [hunk 10]
```diff
@@ -15,18 +15,31 @@ task :first_run do
...
     Rake::Task['graphql:schema:dump'].invoke
   end
 
-  desc 'Dump the schema to IDL in schema.graphql'
+  desc 'Dump the schema to IDL in schema.graphql and generate docs'
   task :idl do
     setup_graphql_rake_tasks
 
     Rake::Task['graphql:schema:idl'].invoke
+    Rake::Task['schema:generate_docs'].invoke
```
Comment (ChristianRimondi) [inline Rakefile:42]: WDYT about generating the docs also in the `schema:dump` and `schema:json`?

File: lib/graphql-docs/landing_pages/index.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
+---
```
Comment (ChristianRimondi) [inline lib/graphql-docs/landing_pages/index.md]: I think that you can remove this file with `graphql-docs` from the PR, right?

File: Rakefile [hunk 12]
```diff
@@ -15,25 +15,40 @@ task :first_run do
...
+    GraphQLDocs.build(
+      filename: "#{__dir__}/schema.graphql",
+      output_dir: 'docs',
+      base_url: '/solidus_graphql_api/docs',
+      landing_pages: { index: "#{doc_dir}/landing_pages/index.md" },
+      delete_output: true
+    )
+  end
+
   desc 'Dump the schema to JSON and IDL'
```
Comment (ChristianRimondi) [inline Rakefile:30]: >[...] and generate docs

File: Rakefile [hunk 13]
```diff
@@ -15,25 +15,40 @@ task :first_run do
...
-  desc 'Dump the schema to IDL in schema.graphql'
+  desc 'Dump the schema to IDL in schema.graphql and generate docs'
   task :idl do
     setup_graphql_rake_tasks
 
     Rake::Task['graphql:schema:idl'].invoke
+    Rake::Task['schema:generate_docs'].invoke
   end
 
   desc 'Dump the schema to JSON in schema.json'
```
Comment (ChristianRimondi) [inline Rakefile:46]: >[...] and generate docs

File: lib/graphql_docs/landing_pages/index.md [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+# Solidus GraphQL API
+
+This is the GraphQL documentation for the [Solidus](https://github.com/solidusio/solidus) ecommerce platform.
+
+## Quick Reference
+
+- Solidus's GraphQL API endpoint is located at `/`.
+- Check out GraphQL's [official documentation](https://graphql.org/)
+
+
```
Comment (ChristianRimondi) [inline lib/graphql_docs/landing_pages/index.md:14]: extra space?

**Pull request merged**